### PR TITLE
Implement T distribution logpdf

### PR DESCRIPTION
**Pull request by fehiepsi**
Currently, gamma sampler (which is the core to make samplers for some other distributions such as Beta/Dirichlet,...)  is not available so I'll defer the implementation of T sampler to a future PR.

Pair code/review with @neerajprad

### CODE REVIEW

File: tests/scipy_stats_test.py [hunk 1]
```diff
@@ -62,6 +62,20 @@ def args_maker():
...
+    lax_fun = lsp_stats.cauchy.logpdf
+
+    def args_maker():
+      x, loc, scale = map(rng, shapes, dtypes)
+      # clipping to ensure that scale is not too low
+      scale = onp.clip(onp.abs(scale), a_min=0.1, a_max=None)
+      return [x, loc, scale]
+
+    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker, check_dtypes=True)
+    self._CompileAndCheck(lax_fun, args_maker, check_dtypes=True)
```
Comment (fehiepsi) [inline tests/scipy_stats_test.py]: This test is reordered for consistency.

**Review by mattjj — APPROVED** [review]
Thanks for this contribution! Super clean as always.

**Comment by mattjj** [comment]
It might make sense to open an "enhancement" issue requesting a gamma sampler. That way we can point other folks at it, like the TF Probability people, to see if they have suggestions.

**Comment by fehiepsi** [comment]
Thanks for your suggestion! I'll do so soon.

**Pull request merged**